tests/apps/pubsub-bulk-subscriber/app.go

```diff
@@ -0,0 +1,577 @@
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/mux"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/dapr/dapr/tests/apps/utils"
+)
+
+const (
+	appPort               = 3000
+	pubsubRawSubTopic     = "pubsub-raw-sub-topic-http"
+	pubsubCESubTopic      = "pubsub-ce-sub-topic-http"
+	pubsubRawBulkSubTopic = "pubsub-raw-bulk-sub-topic-http"
+	pubsubCEBulkSubTopic  = "pubsub-ce-bulk-sub-topic-http"
+	PubSubEnvVar          = "DAPR_TEST_PUBSUB_NAME"
+)
+
+var pubsubkafkaName = "kafka-messagebus"
+
+func init() {
+	if psName := os.Getenv(PubSubEnvVar); len(psName) != 0 {
+		pubsubkafkaName = psName
+	}
+}
+
+type appResponse struct {
+	// Status field for proper handling of errors form pubsub
+	Status    string `json:"status,omitempty"`
+	Message   string `json:"message,omitempty"`
+	StartTime int    `json:"start_time,omitempty"`
+	EndTime   int    `json:"end_time,omitempty"`
+}
+
+type receivedMessagesResponse struct {
+	ReceivedByTopicRawSub     []string `json:"pubsub-raw-sub-topic"`
+	ReceivedByTopicCESub      []string `json:"pubsub-ce-sub-topic"`
+	ReceivedByTopicRawBulkSub []string `json:"pubsub-raw-bulk-sub-topic"`
+	ReceivedByTopicCEBulkSub  []string `json:"pubsub-ce-bulk-sub-topic"`
+}
+
+type subscription struct {
+	PubsubName      string            `json:"pubsubname"`
+	Topic           string            `json:"topic"`
+	Route           string            `json:"route"`
+	DeadLetterTopic string            `json:"deadLetterTopic"`
+	Metadata        map[string]string `json:"metadata"`
+}
+
+type BulkRawMessage struct {
+	Entries  []BulkMessageRawEntry `json:"entries"`
+	Topic    string                `json:"topic"`
+	Metadata map[string]string     `json:"metadata"`
+}
+
+type BulkMessageRawEntry struct {
+	EntryID     string            `json:"entryID"`
+	Event       string            `json:"event"`
+	ContentType string            `json:"contentType,omitempty"`
+	Metadata    map[string]string `json:"metadata"`
+}
+
+type BulkMessage struct {
+	Entries  []BulkMessageEntry `json:"entries"`
+	Topic    string             `json:"topic"`
+	Metadata map[string]string  `json:"metadata"`
+}
+
+type BulkMessageEntry struct {
+	EntryID     string                 `json:"entryID"`
+	Event       map[string]interface{} `json:"event"`
+	ContentType string                 `json:"contentType,omitempty"`
+	Metadata    map[string]string      `json:"metadata"`
+}
+
+type AppBulkMessageEntry struct {
+	EntryID     string            `json:"entryID"`
+	EventStr    string            `json:"event"`
```

Comment:
@DeepanshuA  qq does the app receive the event as a string or is it only in the e2e this is the case ?

Reply:
It is only for the e2e App. Existing e2e tests also used to consider the message as string, I just leveraged existing infrastructure and in my e2e tests as well, user App considers string messages.